Gems/Terrain/Assets/Shaders/Terrain/ClipmapComputeHelpers.azsli

```diff
@@ -188,41 +188,90 @@ struct ClipmapLevel
 
-    float renderDistance = maxRenderDistance / pow(clipmapScaleBase, (float)(clipmapStackSize - 1u - clipmapLevel.m_closestLevel));
-    float blendingStart = renderDistance * BlendingStartFactor;
-    float blendingEnd = renderDistance / RenderDistanceBias;
-    float blendingValue = max(absDistance.x, absDistance.y);
+    float blendBeginDistance = clipmapSize / 2.0 - clipmapMarginSize - ExtendedClipmapMarginSize - BetweenLevelsBlendBandSize;
+    // Clipmap uses Chebyshev distance
+    float2 distance = abs(worldPosition - clipmapCenterInWorldSpace);
+    float cDistance = max(distance.x, distance.y) / clipmapToWorldScale;
+    clipmapLevel.m_weight = max(0.0, (cDistance - blendBeginDistance) / BetweenLevelsBlendBandSize);
 
-    if (blendingValue > blendingStart && blendingValue < blendingEnd)
-    {
-        clipmapLevel.m_weight = (blendingValue - blendingStart) / (blendingEnd - blendingStart);
-        clipmapLevel.m_nextLevel = clipmapLevel.m_closestLevel + 1;
-    }
-    else
+    return clipmapLevel;
+}
+
+ClipmapLevel CalculateDetailClipmapLevel(float2 worldPosition)
+{
+    uint clipmapSize = TerrainSrg::m_clipmapData.m_clipmapSizeUint;
+    uint clipmapMarginSize = TerrainSrg::m_clipmapData.m_detailClipmapMarginSize;
+
+    ClipmapLevel clipmapLevel;
+    float closestLevel = 1.0;
+    // ignore the last level, as it will be used as a fall back default.
+    uint clipmapStackSize = TerrainSrg::m_clipmapData.m_detailClipmapStackSize - 1;
+    for (uint clipmapIndex = 0; clipmapIndex < clipmapStackSize; ++clipmapIndex)
     {
-        clipmapLevel.m_weight = 0.0;
-        clipmapLevel.m_nextLevel = clipmapStackSize;
+        float2 clipmapCenterInWorldSpace = GetDetailClipmapCenterInWorldSpace(clipmapIndex);
+        float clipmapToWorldScale = GetDetailClipmapToWorldScale(clipmapIndex);
+        float extent = (clipmapSize / 2.0 - clipmapMarginSize - ExtendedClipmapMarginSize) * clipmapToWorldScale;
+
+        float contains = AabbContainsPoint(clipmapCenterInWorldSpace, extent, worldPosition);
+        closestLevel += contains;
     }
+    clipmapLevel.m_closestLevel = TerrainSrg::m_clipmapData.m_detailClipmapStackSize - uint(closestLevel);
+    clipmapLevel.m_nextLevel = clipmapLevel.m_closestLevel + 1;
+
+    // Calculate the blending factor if the position falls into the blending band.
+    float2 clipmapCenterInWorldSpace = GetDetailClipmapCenterInWorldSpace(clipmapLevel.m_closestLevel);
+    float clipmapToWorldScale = GetDetailClipmapToWorldScale(clipmapLevel.m_closestLevel);
+
+    float blendBeginDistance = clipmapSize / 2.0 - clipmapMarginSize - ExtendedClipmapMarginSize - BetweenLevelsBlendBandSize;
+    // Clipmap uses Chebyshev distance
+    float2 distance = abs(worldPosition - clipmapCenterInWorldSpace);
+    float cDistance = max(distance.x, distance.y) / clipmapToWorldScale;
+    clipmapLevel.m_weight = max(0.0, (cDistance - blendBeginDistance) / BetweenLevelsBlendBandSize);
```

Comment:
I bet you can barely notice the difference. For close terrian, the first few levels of clipmaps are high resolution anyway; for far away terrain, they are not clearly visible.nside 
It can use Euclidean distance, but that means within a single clipmap, we can only utilize the area inside the incircle of the full texture, in other words 3.14/4 of the area can be used. Then we need to optimize the algorithm to not updating the full clipmap texture.

Reply:
The goal is sampling the best clipmap for the pixel being rendered, not trying to use as much of the high resolution clipmap as possible - if that were true, we wouldn't need mipmaps. This is why mipmaps don't use Chebyshev distance, or even Euclidean distance, they use ddx/ddy of the uvs. Euclidean would be better than Chebyshev because it's at least consistent no matter what direction the camera is pointed.

Reply:
I am not saying that the higher resolution the better. In this particular case, I am saying I don't see a huge gain from switching from Chebyshev distance to Euclidean distance. I believe both performance and visual effect would look similar. Using the above screenshot as an example, you can barely tell where the corner of the square is.
In fact, switching between the two types of distances is trivia. I have no problem switching to that.
But if we want to switch to `SampleGrad()`, it will still involve Chebyshev distance for the level calculation. See discussion here:
https://stackoverflow.com/questions/24388346/how-to-access-automatic-mipmap-level-in-glsl-fragment-shader-texture
Based on [OpenGL 4.2 spec](http://www.opengl.org/registry/doc/glspec42.core.20120427.pdf) chapter 3.9.11 equation 3.21
We can mark this down for future improvement and need to see how it looks like comparing to the current method.

Reply:
In that link, you can see that the mip level should be based on gradients (ddx/ddy), distance to camera really has nothing to do with it, but it works as an approximation for now. Longer term we probably want to go even further and approximate anisotropic filtering which would involve taking multiple samples from different clipmaps and combining them.

Reply:
Yes. I agree with what you say. At some angle the clipmap will show artifacts due to high resolution image. I'll switch to gradient in future PR.